packages/nodecg-utility-obs/src/index.ts

```diff
@@ -158,26 +158,68 @@ export class OBSUtility extends OBSWebSocket {
 		});
 
 		nodecg.listenFor(`${namespace}:transition`, async ({name, duration, sceneName} = {}, callback) => {
-			if (sceneName) {
+			if (studioMode.value) {
+				// If in studio mode, set the preview scene, and then transition to it
+
+				if (sceneName) {
+					try {
+						await this.send('SetPreviewScene', {'scene-name': sceneName});
+					} catch (error) {
+						log.error('Error setting preview scene for transition:', error);
+						if (callback && !callback.handled) {
+							callback(error);
+						}
+						return;
+					}
+				}
+
 				try {
-					await this.send('SetPreviewScene', {'scene-name': sceneName});
+					await this._transition(name, duration);
 				} catch (error) {
-					log.error('Error setting preview scene for transition:', error);
+					log.error('Error transitioning:', error);
 					if (callback && !callback.handled) {
 						callback(error);
 					}
 					return;
 				}
-			}
+			} else {
+				// If not in studio mode, set the transition params and then set the scene
+
+				if (name) {
+					try {
+						await this.send('SetCurrentTransition', { "transition-name": name });
+					} catch (error) {
+						log.error('Error setting scene for transition:', error);
```

Comment:
Should this error instead be something like `Error setting transition name` or `Error setting current transition`?

Reply:
Yup, missed a spot. Thanks for catching it.